pyro/distributions/conjugate.py

```diff
@@ -147,19 +156,9 @@ def sample(self, sample_shape=()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        n = self.total_count
         alpha = self.concentration
-        alpha_sum = self.concentration.sum(-1)
-        if self.is_sparse:
-            value, alpha = torch.broadcast_tensors(value, alpha)
-            elementwise = torch.zeros_like(value)
-            mask = (value != 0)
-            value = value[mask]
-            alpha = alpha[mask]
-            elementwise[mask] = torch.lgamma(value + alpha) - _log_factorial(value) - torch.lgamma(alpha)
-        else:
-            elementwise = torch.lgamma(value + alpha) - _log_factorial(value) - torch.lgamma(alpha)
-        return _log_factorial(n) + torch.lgamma(alpha_sum) - torch.lgamma(n + alpha_sum) + elementwise.sum(-1)
+        return (_log_beta_1(alpha.sum(-1), value.sum(-1), self.is_sparse) -
```

Comment:
Sorry, I need to verify the math at this.

Reply:
@fritzo If I understand correctly, then the reason for using `value.sum(-1)` instead of total_count here is to make it a tensor?

Reply:
The two are identical. We may allow `total_count=None` in the future (as is done by `torch.distributions.Multinomial`), and it's nice to use the `value.sum(-1)` version in anticipation.